### PR TITLE
[Misc] Clean up unused VL, InternVL and Omni model implementations

### PR DESCRIPTION
**Pull request by Joeegin**
## PR Description

Removes model implementations under `vllm_kunlun/models/` that are no longer registered or maintained, together with their `ModelRegistry` entries in `vllm_kunlun/models/__init__.py`.

**What problem does it solve?**
These files were dead code: they are either already served by upstream vLLM, or their `ModelRegistry.register_model(...)` calls had already been commented out, so nothing could reach them at runtime. They still had to be carried, read and kept compiling during refactors.

**Files removed (10, -10771 lines):**

| File | Reason |
| --- | --- |
| `qwen2_vl.py`, `qwen2_5_vl.py` | registration removed; served by upstream vLLM |
| `qwen3_vl.py`, `qwen3_vl_moe.py` | registration already commented out |
| `internvl.py`, `intern_vit.py` | registration removed |
| `interns1.py`, `interns1_vit.py` | registration removed |
| `internlm2.py` | registration removed |
| `qwen3_omni_moe_thinker.py` | never registered, and **already unimportable** — its relative imports `.qwen2_5_vl` / `.qwen3_moe` do not resolve on this branch |

**Approach / verification**

- The diff is pure deletion — 0 added lines.
- All 13 remaining `ModelRegistry` entries in `__init__.py` were checked to resolve to files that still exist (`qwen3_next`, `seed_oss`, `mimo_v2_flash`, `gpt_oss`, `deepseek_v2`, `deepseek_mtp`, `qwen3_5`, `gemma4`, `gemma4_mm`).
- Repo-wide grep for the removed module names: the only remaining hits are `vllm_kunlun/__init__.py` and `vllm_kunlun/models/qwen3_5.py`, which import **upstream** `vllm.model_executor.models.qwen3_vl` — unaffected.
- After deletion the only relative import left under `vllm_kunlun/models/` is `gemma4_mm.py: from .interfaces import SupportsEagle3`; `interfaces.py` is retained. No dangling imports remain.

No functional change, no new tests needed.

---

## Checklist (Required)

- [x] All code changes pass the [`pre-commit`](https://github.com/baidu/vLLM-Kunlun/blob/main/CONTRIBUTING.md) checks.
- [x] Commits are signed off using `git commit -s`.
- [x] The PR title is properly classified (see below).

### CODE REVIEW

File: vllm_kunlun/models/intern_vit.py
Comment (GrootLiu) [inline vllm_kunlun/models/intern_vit.py:1]: internvl.py 使用上游 vllm.model_executor.models.intern_vit

因此这个文件应该本来就是个悬空文件，删除是没有问题的

File: vllm_kunlun/models/internlm2.py [hunk 1]
```diff
@@ -1,470 +0,0 @@
...
-    make_layers,
-    maybe_prefix,
-)
-from vllm.sequence import IntermediateTensors
-
-# from vllm.model_executor.layers.activation import SiluAndMul
-from vllm_kunlun.ops.activation import SiluAndMul
-
-# from vllm.attention import Attention
-from vllm_kunlun.ops.attention.layer import Attention
```
Comment (GrootLiu) [inline vllm_kunlun/models/internlm2.py:50]: 这里依赖了几个 vllm_kunlun 的 ops，直接删除这个文件后会自动使用上游模型组网，因此这里需要确认上游算子和 vllm_kunlun 算子的一致性，这个我之前没看过，麻烦补充一下一致性说明～

**Review by GrootLiu — COMMENTED** [review]
除了 internlm2.py 其他没什么问题了

File: vllm_kunlun/models/internlm2.py (hunk 1 above)
Comment (Joeegin) [reply vllm_kunlun/models/internlm2.py:50]: 感谢 review～排查了一下，这里没有一致性风险，具体如下：

**1. `SiluAndMul`：已经不存在"两份实现"**

`vllm_kunlun/ops/activation.py` 现在只有 39 行，不再提供独立的 `SiluAndMul` 类，而是**原地 monkey-patch 上游的类**：把 `SiluAndMul.forward_native` 换成 `torch.ops._C.silu_and_mul`（→ `kunlun_ops.swiglu`，见 `ops/_custom_ops.py:699-706`），由 `vllm_kunlun/__init__.py:171-186` 的 post-import hook 在 `vllm.model_executor.layers.activation` 被 import 时自动生效。

所以上游 internlm2 组网里 `SiluAndMul()` 构造出来的就是**同一个被 patch 过的类对象**，跑的是同一个昆仑 kernel。dispatch 路径上也是收敛的：`KunlunPlatform._enum = PlatformEnum.OOT` 且 `is_xpu()` / `is_cuda_alike()` 均为 False，`CustomOp.dispatch_forward()` 会选 `forward_oot` → 默认落到 `forward_native`（即 patch 后的实现），不会掉进上游 `forward_xpu` 的 IPEX 分支；即使某条路径落到上游 `forward_cuda`，其内部同样是 `torch.ops._C.silu_and_mul`，而 `vllm._C` 已被 stub 成空模块（`__init__.py:301-303`），该命名空间也归 vllm_kunlun。两条分支殊途同归。

**2. `Attention`：数值路径由平台决定，与模型用哪个 wrapper 无关**

`vllm_kunlun/ops/attention/layer.py` 的 `Attention(VllmAttention)` 只覆写了 `forward`，把输出路径换成 `unified_attention_with_output_kunlun`，而该 op 内部仍是 `self.impl.forward(...)`。真正的算子来自 `self.impl`，即 `KunlunPlatform.get_attn_backend_cls()` 返回的 `KunlunAttentionBackend`（`platforms/kunlun.py:312-315`）—— 这是平台级选择。上游 internlm2 用上游 `Attention` 时，底层跑的仍然是昆仑 backend。wrapper 层的差异仅在于 kunlun 版少了上游的 output_scale / 量化输出分支，且用 `torch.zeros` 而非 `torch.empty` 初始化 output，不影响数值。

**3. 补充一点：这个文件在删除前其实已经 import 不了**

`internlm2.py:47` 的 `from vllm_kunlun.ops.activation import SiluAndMul` 会直接抛 ImportError —— 自 #240 (`b1ebf7a`) 把旧的 `activation.py` 整体删除、后由 `05e4129` 以纯 patch 形式重建之后，该模块就不再导出 `SiluAndMul` 这个名字（只以 `_upstream_cls` 做内部绑定）。所以 `InternLM2ForCausalLM` 虽然注册着，真正加载时会失败。删除后回落到上游组网，反而是修复。

Comment (GrootLiu) [reply vllm_kunlun/models/internlm2.py:50]: 第一个结论不太对，我阅读了 vllm_kunlun/ops/activation.py，里面确实会做 monkey-patch，但是在 vllm_kunlun/\__init\__.py 中没有没有对应的 post-import hook 处理，请再仔细检查一下～

Comment (GrootLiu) [reply vllm_kunlun/models/internlm2.py:50]: 第二点第三点都没问题，再处理一下第一点吧，我感觉需要在 __init__.py 中做个 post-import hook，无论是 vllm_kunlun/\__init\__.py 还是 vllm_kunlun/ops/\__init\__.py，应该都可以

Comment (GrootLiu) [reply vllm_kunlun/models/internlm2.py:50]: 然后 attention.py 中的 _forward_native 名字也不太合理，建议顺手换成 _forward_kulun

Comment (GrootLiu) [reply vllm_kunlun/models/internlm2.py:50]: @YunzhuLu

Comment (Joeegin) [reply vllm_kunlun/models/internlm2.py:50]: 你是对的，我第 1 点的结论有误，已在 44551f1 修复，这里更正一下。

**我错在哪**

我引用的 `vllm_kunlun/__init__.py` 里的 activation post-import hook，是从 `05e4129` 的**提交 diff** 里读的，不是当前文件。这个 hook 在 `ca5c73f`（[Feature] Support Qwen3.6 35B on vLLM 0.25.1，#392）中被删掉了，当前 HEAD 的 hook 5 其实是 `_maybe_get_memory_pool_context`。

**实测确认**

在昆仑机器上验证，修复前：

```
patched? False
forward_native: vllm.model_executor.layers.activation
```

也就是说 `ca5c73f` 之后，`SiluAndMul` 从未被 patch 过，`kunlun_ops.swiglu` 融合算子一直没被用上（走的是上游 torch 原生实现，数值正确但丢了融合 kernel）。你的判断和修复建议都成立。

**44551f1 的修复**

1. 按你的建议把 post-import hook 补回 `vllm_kunlun/__init__.py`（编号顺延为 hook 7，不动现有 1-6）；
2. 顺带在 `vllm_kunlun/ops/activation.py` 补上 `SiluAndMul = _upstream_cls` 的 re-export —— 该模块自 #240 起就不再导出这个名字，导致 `deepseek_v2` / `seed_oss` / `mimo_v2_flash` 里的 `from vllm_kunlun.ops.activation import SiluAndMul` 无法解析。这里导出的就是被 patch 过的上游类本身，不是第二份实现。

修复后实测：

```
patched? True
forward_native: vllm_kunlun.ops.activation
别名 OK: True
```

**一个额外收获**

修复前后对比日志发现，这个 hook 还连带恢复了另外三个 OOT 注册：

```
[KunlunOOT] Registered KunlunRMSNorm and KunlunGemmaRMSNorm via CustomOp.register_oot
[KunlunOOT] Registered KunlunRotaryEmbedding, KunlunMRotaryEmbedding, KunlunDeepseekScalingRotaryEmbedding
[KunlunOOT] Registered KunlunVocabParallelEmbedding
```

因为 `import vllm_kunlun.ops.activation` 会连带加载父包 `vllm_kunlun/ops/__init__.py`。修复前这三行在只加载插件的场景下是不出现的。这个连带行为与 `05e4129` 原版 hook 完全一致，属于回到已知基线，不是新引入的风险面。

**回到本 PR**：删除 `internlm2.py` 的结论不变。vllm_kunlun 侧没有独立的 `SiluAndMul` 实现，patch 打在上游类上，上游组网构造出来的就是同一个类；`Attention` 侧的 impl 由 `KunlunPlatform.get_attn_backend_cls()` 平台级选出，也与模型用哪个 wrapper 无关。

File: vllm_kunlun/ops/activation.py
Comment (GrootLiu) [inline vllm_kunlun/ops/activation.py]: 这里顺便改个名吧，原来的 _forward_native 容易引起歧义，改成 _forward_kunlun 比较好

File: vllm_kunlun/ops/activation.py [hunk 2]
```diff
@@ -36,4 +36,11 @@ def _forward_native(self, x: torch.Tensor) -> torch.Tensor:
...
     _upstream_cls._kunlun_silu_and_mul_patched = True
-    logger.info("[KunlunPlugin] SiluAndMul patched in vllm_kunlun/ops/activations.py")
+    logger.info("[KunlunPlugin] SiluAndMul patched in vllm_kunlun/ops/activation.py")
+
+
+# Re-export so that ``from vllm_kunlun.ops.activation import SiluAndMul`` keeps
+# working for model files written against the pre-patch layout (deepseek_v2,
+# seed_oss, mimo_v2_flash). It is the upstream class itself -- patched above --
+# not a separate implementation.
+SiluAndMul = _upstream_cls
```
Comment (GrootLiu) [inline vllm_kunlun/ops/activation.py]: 这里我有点不理解，这是因为 deepseek_v2 这个模型还使用上游的 SiluAndMul 吗？

File: vllm_kunlun/models/internlm2.py (hunk 1 above)
Comment (YunzhuLu) [reply vllm_kunlun/models/internlm2.py:50]: 看起来当前方案是可以 work 的。
不过，通过 post-import hook 加载/patch 单个算子的实现不太适合作为长期方案。测试中连带恢复其他 OOT 注册虽然回到了原有行为基线，但也说明了目前 OOT 注册依赖隐式的模块导入，并且缺少显式校验；如果相关模块没有按预期加载，可能出现 OOT 漏注册却不能及时发现的情况。
这些问题不阻塞本 PR，后续我们可以再一起梳理算子注册流程，考虑提供统一入口并增加注册结果校验。

**Comment by GrootLiu** [comment]
Hi，这个 PR 最近有什么进展吗

**Comment by Joeegin** [comment]
> 嗨，这个 PR 最近有什么进展吗

没有需要修改的了吧，之前的冲突解决了

**Comment by GrootLiu** [comment]
> > 嗨，这个 PR 最近有什么进展吗
> 
> 没有需要修改的了吧，之前的冲突解决了

还有两个 pending，有空的时候看看～

File: vllm_kunlun/ops/activation.py
Comment (Joeegin) [reply vllm_kunlun/ops/activation.py]: 已改，见 7666811。

只改了模块级函数名 `_forward_native` → `_forward_kunlun`。patch 的**目标属性名仍是 `forward_native`**（第 39 行）—— 那是上游 `CustomOp` 的方法名，OOT dispatch 最终落在它上面，改掉 patch 就失效了：

```python
def _forward_kunlun(self, x): ...

_upstream_cls.forward_native = _forward_kunlun   # 属性名不能动
```

顺手在赋值处加了条注释把这个约束记下来，免得后面有人看到左右名字不一致又去「修正」。纯改名，无行为变化。

File: vllm_kunlun/ops/activation.py (hunk 2 above)
Comment (Joeegin) [reply vllm_kunlun/ops/activation.py]: 是，但更准确的说法是：**现在只剩上游那一个类了**，vllm_kunlun 侧不存在第二份 `SiluAndMul` 实现。

`_upstream_cls` 就是第 22 行 import 进来的上游类，第 39 行是**原地改它的类属性**。所以 `SiluAndMul = _upstream_cls` 导出的是同一个已被 patch 过的对象 —— `from vllm.model_executor.layers.activation import SiluAndMul` 和 `from vllm_kunlun.ops.activation import SiluAndMul` 拿到的东西 `is` 判等为 True，整个进程只有这一个类。

加这个别名纯粹是为了把**断掉的 import 接回去**。自 #240 起该模块不再导出 `SiluAndMul` 这个名字，而这三处一直按老布局写：

- `vllm_kunlun/models/deepseek_v2.py:84`
- `vllm_kunlun/models/seed_oss.py:61`
- `vllm_kunlun/models/mimo_v2_flash.py:46`

也就是说在补上别名之前，这三行 `from vllm_kunlun.ops.activation import SiluAndMul` 是解析不了的。补上之后它们拿到的仍然是跑 `kunlun_ops.swiglu` 的那个类（`torch.ops._C.silu_and_mul` → `ops/_custom_ops.py:699-706`，且 `vllm._C` / `vllm._moe_C` 已被 stub，该命名空间归 vllm_kunlun），不是回退到上游 torch 实现。

如果你觉得这个别名容易引起「有两份实现」的误会，另一个方向是反过来改那三个模型文件、直接 import 上游路径，别名就可以删掉。不过那会动到本 PR 范围外的文件，我倾向留到后续单独处理——听你的。

**Comment by Joeegin** [comment]
我又仔细看了下，我感觉可以改成OOT的形式，这块我再看看

File: vllm_kunlun/ops/activation.py (hunk 2 above)
Comment (Joeegin) [reply vllm_kunlun/ops/activation.py]: 我该写成OOT形式了

**Comment by Joeegin** [comment]
已更新成OOT形式

**Review by xyDong0223 — APPROVED** [review]

**Comment by xyDong0223** [comment]
Thanks

**Comment by GrootLiu** [comment]
> 已更新成OOT形式

目前 mimo v2 flash imagezh这个部分应该也可以直接改成 from vllm.model_executor.layers.attention import Attention，完全剔除从  vllm kunlun import 内容让
其他两个模型（deepseek v2, seed oss）应该也能这样做，这个新提一个 PR 吧～

**Comment by GrootLiu** [comment]
> > 已更新成OOT形式
> 
> 目前 mimo v2 flash imagezh这个部分应该也可以直接改成 from vllm.model_executor.layers.attention import Attention，完全剔除从 vllm kunlun import 内容让 其他两个模型（deepseek v2, seed oss）应该也能这样做，这个新提一个 PR 吧～

@feiyu-byte
